<a href="https://colab.research.google.com/github/kittikhun62/DQ/blob/main/midterm.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
unit = pd.read_csv('/content/drive/MyDrive/BPDV65/opendata_unittype.csv')
unit.head()

row_number     unittype_id    project_id             name_th  \
0       27501   unittype_9565  project_2446           2 Bedroom   
1       28777  unittype_10267  project_3543  บ้านแฝดสองชั้น Z-2   
2       28865    unittype_471   project_143    1 Bedroom Type B   
3       24988  unittype_19506  project_7766    Studio Floor 1-2   
4       38596  unittype_10290  project_3551                TH 3   

   propertytype_id propertytype_name_en propertytype_name_th  \
0                2                Condo                คอนโด   
1            20000           Twin house              บ้านแฝด   
2                2                Condo                คอนโด   
3                2                Condo                คอนโด   
4                3             Townhome             ทาวน์โฮม   

   area_total_min_wa  area_usable_min  price_min  ...  count_room_guest  \
0                NaN             37.3  2500000.0  ...               1.0   
1               40.0            150.0  2390000.0  ...               NaN   
2                NaN             28.0        NaN  ...               NaN   
3                NaN             27.0   170000.0  ...               NaN   
4                NaN            182.0  2390000.0  ...               1.0   

   count_room_kitchen  count_room_living  count_room_maid  count_room_storage  \
0                 1.0                NaN              NaN                 NaN   
1                 NaN                NaN              NaN                 NaN   
2                 NaN                NaN              NaN                 NaN   
3                 NaN                NaN              NaN                 NaN   
4                 1.0                NaN              NaN                 2.0   

   count_room_utility         date_created         date_updated  source  \
0                 NaN  2021-11-19 18:00:09  2021-11-19 17:44:45  baania   
1                 NaN  2022-01-27 14:30:04  2022-01-27 14:22:55  baania   
2                 NaN  2022-01-14 15:00:10  2022-01-14 14:40:04  baania   
3                 NaN  2020-12-04 10:40:59  2020-10-14 01:06:48  baania   
4                 NaN  2021-11-12 13:30:08  2021-02-03 12:12:54  baania   

                                        url_unittype  
0  https://baania.com/th/project/เฟ-ลิค-คอนโด-ลาด...  
1  https://baania.com/th/project/เซน-บาย-ศิวาลัย-...  
2  https://baania.com/th/project/วันพลัส-นายน์ทีน...  
3  https://baania.com/th/project/บ้านสวนชมน้ำ-776...  
4  https://baania.com/th/project/เบล็สทาวน์-ศรีนค...  

[5 rows x 27 columns]

In [4]:
project = pd.read_csv('/content/drive/MyDrive/BPDV65/opendata_project.csv')
project.head()

row_number                project_id                             name_en  \
0      7880.0  5ee08d64dda791001299b2a6  Winning Residence Sukhumvit Praksa   
1     16689.0             project_11326                    Gift Condominium   
2       290.0  62739c4089233e001362b001    Sanrak 9 Sukprayoon Nong Krathum   
3     20719.0              project_2010                     Baan Sansabai 1   
4       199.0             project_16838          Baanpruksa Ratchapruek-345   

                              name_th propertytype_id propertytype_name_en  \
0  วินนิ่ง เรสซิเดนท์ สุขุมวิท แพรกษา               1       Detached House   
1                 กิ๊ฟท์ คอนโดมิเนียม               2                Condo   
2      แสนรัก 9 ศุขประยูร หนองกระทุ่ม               3             Townhome   
3                       บ้านแสนสบาย 1               1       Detached House   
4              บ้านพฤกษา ราชพฤกษ์-345               3             Townhome   

  propertytype_name_th  price_min     developer_id  \
0                 บ้าน  3790000.0  developer_10678   
1                คอนโด  1100000.0   developer_7319   
2             ทาวน์โฮม        0.0  developer_10510   
3                 บ้าน        0.0  developer_10510   
4             ทาวน์โฮม  1930000.0   developer_1026   

              developer_name_en  ... facility_meeting  facility_park  \
0  Winning Asset Plus Co., Ltd.  ...              NaN            NaN   
1      Gift Asset Land Co.,Ltd.  ...              NaN            1.0   
2                           NaN  ...              NaN            NaN   
3                           NaN  ...              NaN            NaN   
4  Preuksa Real Estate Co.,Ltd.  ...              NaN            1.0   

   facility_playground facility_pool facility_security         date_created  \
0                  NaN           1.0               1.0  2020-06-10 14:36:04   
1                  NaN           1.0               1.0  2017-10-30 15:38:48   
2                  NaN           NaN               NaN  2022-05-05 16:43:28   
3                  NaN           NaN               NaN  2016-09-05 15:32:15   
4                  NaN           1.0               1.0  2019-09-05 11:49:37   

           date_finish         date_updated  source  \
0                  NaN  2020-12-04 10:32:58  baania   
1  2013-01-01 07:00:00  2020-12-04 10:49:00  baania   
2                  NaN  2022-05-05 17:30:03  baania   
3  1970-01-02 07:00:00  2020-12-04 10:45:31  baania   
4  1970-01-02 07:00:00  2021-03-17 12:00:04  baania   

                                         url_project  
0  https://baania.com/th/project/วินนิ่ง-เรสซิเดน...  
1  https://baania.com/th/project/กิ๊ฟท์-คอนโดมิเน...  
2  https://baania.com/th/project/แสนรัก-9-ศุขประย...  
3  https://baania.com/th/project/บ้านแสนสบาย-1-pr...  
4  https://baania.com/th/project/บ้านพฤกษา-ราชพฤก...  

[5 rows x 45 columns]

In [5]:
rent = pd.read_csv('/content/drive/MyDrive/BPDV65/opendata_volume_listing_rent.csv')
rent.head()

row_number  transction_year  transction_month listing_province_name_th  \
0        1324             2020                12            กรุงเทพมหานคร   
1         136             2020                10            กรุงเทพมหานคร   
2         310             2020                 5                  นนทบุรี   
3        1570             2020                 3            กรุงเทพมหานคร   
4        1063             2019                 7            กรุงเทพมหานคร   

  listing_province_name_en listing_district_name_th listing_district_name_en  \
0                  BANGKOK                  จตุจักร                CHATUCHAK   
1                  BANGKOK               บางกอกน้อย              BANGKOK NOI   
2               NONTHABURI                 ปากเกร็ด                 PAK KRET   
3                  BANGKOK                   บางรัก                 BANG RAK   
4                  BANGKOK                   พระนคร              PHRA NAKHON   

  project_propertytype_name_th project_propertytype_name_en  volume_of_rental  \
0                        คอนโด                        Condo               702   
1                        คอนโด                        Condo                33   
2                        คอนโด                        Condo                22   
3                        คอนโด                        Condo               104   
4                        คอนโด                        Condo                11   

          day  
0  2022-08-31  
1  2022-08-31  
2  2022-08-31  
3  2022-08-31  
4  2022-08-31

In [6]:
eat = pd.read_csv('/content/drive/MyDrive/BPDV65/opendata_eating_score.csv')
eat.head()

row_number  subdistrict_id subdistrict_name_en subdistrict_name_th  \
0           2            4001         HUAI KHWANG            ห้วยขวาง   
1         397            2632      THEP KRASATTRI         เทพกระษัตรี   
2          76            3509             PAK NAM              ปากน้ำ   
3         368            4628               SAMET               เสม็ด   
4         531            1166             TON PAO              ต้นเปา   

   district_id     district_name_en  district_name_th  province_id  \
0         4000          HUAI KHWANG          ห้วยขวาง         3781   
1         2630              THALANG              ถลาง         2616   
2         3499  MUEANG SAMUT PRAKAN  เมืองสมุทรปราการ         3498   
3         4617     MUEANG CHON BURI       เมืองชลบุรี         4616   
4         1164        SAN KAMPHAENG          สันกำแพง         1005   

  province_name_en province_name_th  eating_daytime  eating_daytime_price  \
0          Bangkok    กรุงเทพมหานคร            66.0                  72.0   
1           Phuket           ภูเก็ต            35.0                  37.0   
2     Samut Prakan      สมุทรปราการ            70.0                  84.0   
3        Chon Buri           ชลบุรี            68.0                  67.0   
4       Chiang Mai        เชียงใหม่            44.0                  36.0   

   eating_daytime_quality  eating_daytime_variety  eating_nighttime  \
0                    49.0                    92.0              75.0   
1                    30.0                    45.0              38.0   
2                    55.0                    79.0              65.0   
3                    56.0                    97.0              69.0   
4                    47.0                    51.0              47.0   

   eating_nighttime_price  eating_nighttime_quality  eating_nighttime_variety  \
0                    68.0                      54.0                      94.0   
1                    28.0                      28.0                      51.0   
2                    72.0                      57.0                      67.0   
3                    41.0                      54.0                      94.0   
4                    22.0                      61.0                      52.0   

   source         day  
0  baania  2022-08-31  
1  baania  2022-08-31  
2  baania  2022-08-31  
3  baania  2022-08-31  
4  baania  2022-08-31

#Midterm Questions เก็บ 20 คะแนน
1. หาพื้นที่ (district_name)ที่มีจำนวนโครงการเยอะที่สุดและมีความต้องการเช่าเยอะที่สุด (5 คะแนน)
2. โครงการที่แพงที่สุดในพื้นที่ที่มีความต้องการเช่าสูงที่สุด (5 คะแนน)
3. ราคาเฉลี่ยของโครงการ ที่อยู่ในย่าน (subdistrict) ที่ อาหารการกินมีความหลากหลายที่สุดในช่วงกลางวัน (eating_daytime_variety) ของประเทศไทย (5 คะแนน)
4. พื้นที่ ที่มีความต้องการเช่ามากที่สุด คือพื้นที่ใด และพื้นที่นั้นมีชนิดของบ้าน แบบไหนมากที่สุด (7 คะแนน)
5. (เฉพาะ กรุงเทพมหานคร) เปรียบเทียบราคาโครงการที่แพงที่สุดในย่าน (subdistrict)ที่ราคาของอาหารการกินในช่วงกลางคืนที่ราคาสูงที่สุด (eating_nighttime_price) กับ ย่าน (subdistrict)ที่ราคาของอาหารการกินในช่วงกลางคืนที่ราคาต่ำที่สุด (eating_nighttime_price) (10 คะแนน)
6. แสดงย่านที่ อาหารการกินมีความหลากหลายที่สุดในช่วงกลางวัน (eating_daytime_variety) และ กลางคืน (eating_nighttime_variety) ของแต่ละจังหวัด (13 คะแนน)

In [7]:
unit.isnull().any()

row_number              False
unittype_id             False
project_id              False
name_th                  True
propertytype_id         False
propertytype_name_en    False
propertytype_name_th    False
area_total_min_wa        True
area_usable_min          True
price_min                True
count_airconditioner     True
count_elevator           True
count_floor_internal     True
count_parking            True
count_room_bath          True
count_room_bed           True
count_room_dinning       True
count_room_guest         True
count_room_kitchen       True
count_room_living        True
count_room_maid          True
count_room_storage       True
count_room_utility       True
date_created            False
date_updated            False
source                  False
url_unittype             True
dtype: bool

In [8]:
project.isnull().any()

row_number                 True
project_id                False
name_en                   False
name_th                   False
propertytype_id           False
propertytype_name_en      False
propertytype_name_th      False
price_min                  True
developer_id              False
developer_name_en          True
developer_name_th          True
latitude                  False
longitude                  True
neighborhood_id            True
neighborhood_name_en       True
neighborhood_name_th       True
subdistrict_id             True
subdistrict_name_en        True
subdistrict_name_th        True
district_id                True
district_name_en           True
district_name_th           True
province_id                True
province_name_en           True
province_name_th           True
zipcode                    True
count_elevator             True
count_elevator_service     True
count_floor                True
count_parking              True
count_tower                True
count_un

In [9]:
rent.isnull().any()

row_number                      False
transction_year                 False
transction_month                False
listing_province_name_th        False
listing_province_name_en        False
listing_district_name_th        False
listing_district_name_en        False
project_propertytype_name_th    False
project_propertytype_name_en    False
volume_of_rental                False
day                             False
dtype: bool

In [10]:
eat.isnull().any()

row_number                  False
subdistrict_id              False
subdistrict_name_en         False
subdistrict_name_th         False
district_id                 False
district_name_en            False
district_name_th            False
province_id                 False
province_name_en            False
province_name_th            False
eating_daytime              False
eating_daytime_price        False
eating_daytime_quality      False
eating_daytime_variety      False
eating_nighttime            False
eating_nighttime_price      False
eating_nighttime_quality    False
eating_nighttime_variety    False
source                      False
day                         False
dtype: bool

In [11]:
unit1=unit.dropna(subset=['row_number','unittype_id','project_id','propertytype_id','propertytype_name_en','propertytype_name_th'])

In [12]:
project1=project.dropna(subset=['project_id','name_th','propertytype_id','propertytype_name_th','developer_id',])

In [13]:
rent1=rent.dropna(subset=['row_number','transction_year','transction_month','listing_province_name_th','listing_district_name_th','project_propertytype_name_th','volume_of_rental'])

In [14]:
eat1=eat.dropna(subset=['row_number','subdistrict_id','subdistrict_name_th','district_id','district_name_th','province_id','province_name_th','eating_daytime','eating_daytime_price','eating_daytime_variety','eating_nighttime_variety'])

1.หาพื้นที่ (district_name)ที่มีจำนวนโครงการเยอะที่สุดและมีความต้องการเช่าเยอะที่สุด (5 คะแนน)

In [15]:
dataall = project1.merge(rent1, left_on='row_number', right_on='row_number')
dataall.head()

row_number                project_id                               name_en  \
0       290.0  62739c4089233e001362b001      Sanrak 9 Sukprayoon Nong Krathum   
1       199.0             project_16838            Baanpruksa Ratchapruek-345   
2       270.0  5efc3cb4bc2af40019402f74     Golden Valley Assumption-Sriracha   
3      1432.0             project_10000                      Baan Boonraksa 7   
4       679.0             project_12336  Baan Klangmeung Urbanion Ladphrao 71   

                                   name_th propertytype_id  \
0           แสนรัก 9 ศุขประยูร หนองกระทุ่ม               3   
1                   บ้านพฤกษา ราชพฤกษ์-345               3   
2        โกลเด้น วัลเลย์ อัสสัมชัญ-ศรีราชา               1   
3                           บ้านบุญรักษา 7               1   
4  บ้านกลางเมือง เออร์บาเนี่ยน ลาดพร้าว 71               6   

  propertytype_name_en propertytype_name_th  price_min     developer_id  \
0             Townhome             ทาวน์โฮม        0.0  developer_10510   
1             Townhome             ทาวน์โฮม  1930000.0   developer_1026   
2       Detached House                 บ้าน  2590000.0  developer_10510   
3       Detached House                 บ้าน  1190000.0   developer_7556   
4          Home office            โฮมออฟฟิศ  6500000.0   developer_7953   

              developer_name_en  ... transction_year  transction_month  \
0                           NaN  ...            2020                 5   
1  Preuksa Real Estate Co.,Ltd.  ...            2019                10   
2                           NaN  ...            2021                 2   
3        BOONKERDKIAT CO., LTD.  ...            2020                 4   
4                           NaN  ...            2020                 5   

   listing_province_name_th listing_province_name_en listing_district_name_th  \
0             กรุงเทพมหานคร                  BANGKOK               บางกอกใหญ่   
1             กรุงเทพมหานคร                  BANGKOK                   ประเวศ   
2             กรุงเทพมหานคร                  BANGKOK                  สวนหลวง   
3             กรุงเทพมหานคร                  BANGKOK                   บางรัก   
4             กรุงเทพมหานคร                  BANGKOK                ภาษีเจริญ   

  listing_district_name_en  project_propertytype_name_th  \
0              BANGKOK YAI                         คอนโด   
1                   PRAWET                         คอนโด   
2             PHATTHANAKAN                         คอนโด   
3                 BANG RAK                         คอนโด   
4            PHASI CHAROEN                         คอนโด   

  project_propertytype_name_en volume_of_rental         day  
0                        Condo               11  2022-08-31  
1                        Condo              169  2022-08-31  
2                        Condo               58  2022-08-31  
3                        Condo             1230  2022-08-31  
4                        Condo               42  2022-08-31  

[5 rows x 55 columns]

In [16]:
dataall2 =dataall.merge(eat1, left_on='row_number', right_on='row_number')
dataall2.describe()

row_number    latitude   longitude  subdistrict_id_x  district_id_x  \
count  554.000000  554.000000  554.000000        553.000000     554.000000   
mean   277.500000   13.673928  100.613183       4201.819168    4072.577617   
std    160.070297    2.446445    1.017573       2242.850658    1749.072333   
min      1.000000    6.912465   98.299281       1008.000000    1006.000000   
25%    139.250000   13.416863  100.406451       3424.000000    3418.000000   
50%    277.500000   13.764385  100.615980       3803.000000    3800.000000   
75%    415.750000   13.956280  100.890689       4527.000000    4511.000000   
max    554.000000   20.438575  104.860601      20624.000000   10697.000000   

       province_id_x       zipcode   count_unit  count_unittype  \
count     554.000000    554.000000   503.000000      402.000000   
mean     4009.732852  29297.743682   196.075547        2.253731   
std      1751.350598  25749.275447   302.923889        1.504871   
min      1005.000000  10110.000000     2.000000        1.000000   
25%      3372.000000  10510.000000    43.500000        1.000000   
50%      3781.000000  20000.000000    97.000000        2.000000   
75%      4510.000000  43500.000000   236.000000        3.000000   
max     10508.000000  92000.000000  3280.000000        8.000000   

       facility_clubhouse  ...  district_id_y  province_id_y  eating_daytime  \
count          164.000000  ...     554.000000     554.000000      554.000000   
mean             0.780488  ...    3456.799639    3393.846570       54.949458   
std              0.415184  ...     950.806754     963.250853       14.018238   
min              0.000000  ...    1006.000000    1005.000000        0.000000   
25%              1.000000  ...    3432.000000    3431.000000       45.250000   
50%              1.000000  ...    3647.000000    3599.000000       58.000000   
75%              1.000000  ...    3895.000000    3781.000000       66.000000   
max              1.000000  ...    4717.000000    4616.000000       91.000000   

       eating_daytime_price  eating_daytime_quality  eating_daytime_variety  \
count            554.000000              554.000000              554.000000   
mean              63.167870               45.967509               60.725632   
std               19.182449               15.853729               22.390271   
min                0.000000                0.000000                0.000000   
25%               49.000000               35.250000               45.500000   
50%               67.000000               48.000000               63.500000   
75%               80.000000               57.000000               77.000000   
max               95.000000               93.000000              100.000000   

       eating_nighttime  eating_nighttime_price  eating_nighttime_quality  \
count        554.000000              554.000000                554.000000   
mean          54.608303               55.258123                 43.796029   
std           15.153584               20.479429                 17.302748   
min            5.000000                0.000000                  0.000000   
25%           44.000000               40.000000                 31.000000   
50%           57.000000               59.000000                 45.000000   
75%           66.000000               73.000000                 56.000000   
max           82.000000              100.000000                 79.000000   

       eating_nighttime_variety  
count                554.000000  
mean                  61.386282  
std                   21.826346  
min                    0.000000  
25%                   47.000000  
50%                   64.000000  
75%                   77.000000  
max                  100.000000  

[8 rows x 30 columns]

In [17]:
dataall2.groupby("district_name_th_y")[["name_th"]].max().sort_index(ascending=False) #_y = ชื่อพื้นที่ทั่วประเทศ ทีมีโครงการเยอะสุด
ans_1_1 = dataall2.groupby("district_name_th_y")[["name_th"]].max().reset_index().sort_index(ascending=False) 
ans_1_1.iloc[[0]]

district_name_th_y             name_th
105            ไทรน้อย  ไพร์ม สแควร์ ชุมแพ

In [18]:
dataall2.groupby("volume_of_rental")[["district_name_th_y"]].max().sort_index(ascending=False) #_y = ชื่อพื้นที่ทั่วประเทศ ทีมีโครงการเยอะสุด
ans_1_2 = dataall2.groupby("volume_of_rental")[["district_name_th_y"]].max().sort_index(ascending=False) 
ans_1_2.iloc[[0]]

district_name_th_y
volume_of_rental                   
3996                  เมืองปทุมธานี

In [19]:
print(f'พื้นที่ที่มีโครงการเยอะที่สุด คือ {ans_1_1.iloc[0,0]} โครงการที่เยอะที่สุดคือ {ans_1_1.iloc[0,1]}')
print(f'แต่พื้นที่ที่มีความต้องการเช่ามากที่สุด คือ {ans_1_2.iloc[0,0]}')

พื้นที่ที่มีโครงการเยอะที่สุด คือ ไทรน้อย โครงการที่เยอะที่สุดคือ ไพร์ม สแควร์ ชุมแพ
แต่พื้นที่ที่มีความต้องการเช่ามากที่สุด คือ เมืองปทุมธานี


3. ราคาเฉลี่ยของโครงการ ที่อยู่ในย่าน (subdistrict) ที่ อาหารการกินมีความหลากหลายที่สุดในช่วงกลางวัน (eating_daytime_variety) ของประเทศไทย (5 คะแนน)



In [20]:
eat

row_number  subdistrict_id subdistrict_name_en subdistrict_name_th  \
0             2            4001         HUAI KHWANG            ห้วยขวาง   
1           397            2632      THEP KRASATTRI         เทพกระษัตรี   
2            76            3509             PAK NAM              ปากน้ำ   
3           368            4628               SAMET               เสม็ด   
4           531            1166             TON PAO              ต้นเปา   
..          ...             ...                 ...                 ...   
549         116            3726          THA TAMNAK           ท่าตำหนัก   
550         151            3476            CHAK BOK               ชากบก   
551         290            1015             MAE HIA            แม่เหียะ   
552          69            1020      NONG PA KHRANG        หนองป่าครั่ง   
553          35            1165          CHAE CHANG             แช่ช้าง   

     district_id     district_name_en  district_name_th  province_id  \
0           4000          HUAI KHWANG          ห้วยขวาง         3781   
1           2630              THALANG              ถลาง         2616   
2           3499  MUEANG SAMUT PRAKAN  เมืองสมุทรปราการ         3498   
3           4617     MUEANG CHON BURI       เมืองชลบุรี         4616   
4           1164        SAN KAMPHAENG          สันกำแพง         1005   
..           ...                  ...               ...          ...   
549         3719       NAKHON CHAI SI         นครชัยศรี         3667   
550         3474             BAN KHAI          บ้านค่าย         3431   
551         1006    MUEANG CHIANG MAI    เมืองเชียงใหม่         1005   
552         1006    MUEANG CHIANG MAI    เมืองเชียงใหม่         1005   
553         1164        SAN KAMPHAENG          สันกำแพง         1005   

    province_name_en province_name_th  eating_daytime  eating_daytime_price  \
0            Bangkok    กรุงเทพมหานคร            66.0                  72.0   
1             Phuket           ภูเก็ต            35.0                  37.0   
2       Samut Prakan      สมุทรปราการ            70.0                  84.0   
3          Chon Buri           ชลบุรี            68.0                  67.0   
4         Chiang Mai        เชียงใหม่            44.0                  36.0   
..               ...              ...             ...                   ...   
549    Nakhon Pathom           นครปฐม            65.0                  68.0   
550           Rayong            ระยอง            44.0                  42.0   
551       Chiang Mai        เชียงใหม่            66.0                  58.0   
552       Chiang Mai        เชียงใหม่            69.0                  58.0   
553       Chiang Mai        เชียงใหม่            34.0                  38.0   

     eating_daytime_quality  eating_daytime_variety  eating_nighttime  \
0                      49.0                    92.0              75.0   
1                      30.0                    45.0              38.0   
2                      55.0                    79.0              65.0   
3                      56.0                    97.0              69.0   
4                      47.0                    51.0              47.0   
..                      ...                     ...               ...   
549                    56.0                    76.0              64.0   
550                    53.0                    27.0              38.0   
551                    68.0                    72.0              64.0   
552                    67.0                    91.0              72.0   
553                    35.0                    26.0              32.0   

     eating_nighttime_price  eating_nighttime_quality  \
0                      68.0                      54.0   
1                      28.0                      28.0   
2                      72.0                      57.0   
3                      41.0                      54.0   
4                      22.0                      61.0   
..                      ...                       ...   
549           

In [21]:
x = eat[['subdistrict_name_th','eating_daytime_variety','eating_daytime_price']]
x

subdistrict_name_th  eating_daytime_variety  eating_daytime_price
0              ห้วยขวาง                    92.0                  72.0
1           เทพกระษัตรี                    45.0                  37.0
2                ปากน้ำ                    79.0                  84.0
3                 เสม็ด                    97.0                  67.0
4                ต้นเปา                    51.0                  36.0
..                  ...                     ...                   ...
549           ท่าตำหนัก                    76.0                  68.0
550               ชากบก                    27.0                  42.0
551            แม่เหียะ                    72.0                  58.0
552        หนองป่าครั่ง                    91.0                  58.0
553             แช่ช้าง                    26.0                  38.0

[554 rows x 3 columns]

In [22]:
 x.groupby('subdistrict_name_th').mean() # หาราคาเฉลี่ย ของ subdistrict 

eating_daytime_variety  eating_daytime_price
subdistrict_name_th                                              
กมลา                                   48.0                  28.0
กระทุ่มราย                             45.0                  87.0
กระทุ่มล้ม                             84.0                  64.0
กระแชง                                 60.0                  39.0
กระแสบน                                35.0                  27.0
...                                     ...                   ...
ไทรน้อย                                30.0                  82.0
ไทรม้า                                 68.0                  73.0
ไม้ขาว                                 33.0                  35.0
ไร่ขิง                                 70.0                  63.0
ไร่หลักทอง                             38.0                  45.0

[518 rows x 2 columns]

In [23]:
 z = x.groupby('subdistrict_name_th')[['eating_daytime_price']].mean()  
z.sort_values('eating_daytime_price',ascending=1) # เรียงราคาเฉลี่ยจากมากไปน้อย 


eating_daytime_price
subdistrict_name_th                      
ท้ายเกาะ                              0.0
บ้านยาง                               0.0
หนองไผ่แก้ว                          12.0
คลองกิ่ว                             15.0
นาตาขวัญ                             15.0
...                                   ...
หนองแขม                              90.0
พิมลราช                              93.0
บางโปรง                              94.0
พระปฐมเจดีย์                         95.0
ลำต้อยติ่ง                           95.0

[518 rows x 1 columns]

4. พื้นที่ ที่มีความต้องการเช่ามากที่สุด คือพื้นที่ใด และพื้นที่นั้นมีชนิดของบ้าน แบบไหนมากที่สุด (7 คะแนน)

In [24]:
rent

row_number  transction_year  transction_month listing_province_name_th  \
0           1324             2020                12            กรุงเทพมหานคร   
1            136             2020                10            กรุงเทพมหานคร   
2            310             2020                 5                  นนทบุรี   
3           1570             2020                 3            กรุงเทพมหานคร   
4           1063             2019                 7            กรุงเทพมหานคร   
...          ...              ...               ...                      ...   
1873        1585             2019                 8            กรุงเทพมหานคร   
1874         351             2019                12            กรุงเทพมหานคร   
1875         694             2020                 7                  นนทบุรี   
1876         633             2020                 2            กรุงเทพมหานคร   
1877         899             2022                 1                 ปทุมธานี   

     listing_province_name_en listing_district_name_th  \
0                     BANGKOK                  จตุจักร   
1                     BANGKOK               บางกอกน้อย   
2                  NONTHABURI                 ปากเกร็ด   
3                     BANGKOK                   บางรัก   
4                     BANGKOK                   พระนคร   
...                       ...                      ...   
1873                  BANGKOK                บางคอแหลม   
1874                  BANGKOK                  ราชเทวี   
1875               NONTHABURI             เมืองนนทบุรี   
1876                  BANGKOK                  หลักสี่   
1877             PATHUM THANI                  ลำลูกกา   

     listing_district_name_en project_propertytype_name_th  \
0                   CHATUCHAK                        คอนโด   
1                 BANGKOK NOI                        คอนโด   
2                    PAK KRET                        คอนโด   
3                    BANG RAK                        คอนโด   
4                 PHRA NAKHON                        คอนโด   
...                       ...                          ...   
1873            BANG KHO LAEM                        คอนโด   
1874              RATCHATHEWI                        คอนโด   
1875        MUEANG NONTHABURI                        คอนโด   
1876                   LAK SI                        คอนโด   
1877               LAM LUK KA                        คอนโด   

     project_propertytype_name_en  volume_of_rental         day  
0                           Condo               702  2022-08-31  
1                           Condo                33  2022-08-31  
2                           Condo                22  2022-08-31  
3                           Condo               104  2022-08-31  
4                           Condo                11  2022-08-31  
...                           ...               ...         ...  
1873                        Condo                23  2022-08-31  
1874                        Condo               103  2022-08-31  
1875                        Condo               100  2022-08-31  
1876                        Condo                14  2022-08-31  
1877                        Condo                23  2022-08-31  

[1878 rows x 11 columns]

In [25]:
rent


row_number  transction_year  transction_month listing_province_name_th  \
0           1324             2020                12            กรุงเทพมหานคร   
1            136             2020                10            กรุงเทพมหานคร   
2            310             2020                 5                  นนทบุรี   
3           1570             2020                 3            กรุงเทพมหานคร   
4           1063             2019                 7            กรุงเทพมหานคร   
...          ...              ...               ...                      ...   
1873        1585             2019                 8            กรุงเทพมหานคร   
1874         351             2019                12            กรุงเทพมหานคร   
1875         694             2020                 7                  นนทบุรี   
1876         633             2020                 2            กรุงเทพมหานคร   
1877         899             2022                 1                 ปทุมธานี   

     listing_province_name_en listing_district_name_th  \
0                     BANGKOK                  จตุจักร   
1                     BANGKOK               บางกอกน้อย   
2                  NONTHABURI                 ปากเกร็ด   
3                     BANGKOK                   บางรัก   
4                     BANGKOK                   พระนคร   
...                       ...                      ...   
1873                  BANGKOK                บางคอแหลม   
1874                  BANGKOK                  ราชเทวี   
1875               NONTHABURI             เมืองนนทบุรี   
1876                  BANGKOK                  หลักสี่   
1877             PATHUM THANI                  ลำลูกกา   

     listing_district_name_en project_propertytype_name_th  \
0                   CHATUCHAK                        คอนโด   
1                 BANGKOK NOI                        คอนโด   
2                    PAK KRET                        คอนโด   
3                    BANG RAK                        คอนโด   
4                 PHRA NAKHON                        คอนโด   
...                       ...                          ...   
1873            BANG KHO LAEM                        คอนโด   
1874              RATCHATHEWI                        คอนโด   
1875        MUEANG NONTHABURI                        คอนโด   
1876                   LAK SI                        คอนโด   
1877               LAM LUK KA                        คอนโด   

     project_propertytype_name_en  volume_of_rental         day  
0                           Condo               702  2022-08-31  
1                           Condo                33  2022-08-31  
2                           Condo                22  2022-08-31  
3                           Condo               104  2022-08-31  
4                           Condo                11  2022-08-31  
...                           ...               ...         ...  
1873                        Condo                23  2022-08-31  
1874                        Condo               103  2022-08-31  
1875                        Condo               100  2022-08-31  
1876                        Condo                14  2022-08-31  
1877                        Condo                23  2022-08-31  

[1878 rows x 11 columns]

เลือกคอลัมที่ต้องการ

In [26]:
Rent_0 = rent[['row_number','listing_district_name_th','volume_of_rental']]
Rent_0


row_number listing_district_name_th  volume_of_rental
0           1324                  จตุจักร               702
1            136               บางกอกน้อย                33
2            310                 ปากเกร็ด                22
3           1570                   บางรัก               104
4           1063                   พระนคร                11
...          ...                      ...               ...
1873        1585                บางคอแหลม                23
1874         351                  ราชเทวี               103
1875         694             เมืองนนทบุรี               100
1876         633                  หลักสี่                14
1877         899                  ลำลูกกา                23

[1878 rows x 3 columns]

In [27]:
project

row_number                project_id  \
0          7880.0  5ee08d64dda791001299b2a6   
1         16689.0             project_11326   
2           290.0  62739c4089233e001362b001   
3         20719.0              project_2010   
4           199.0             project_16838   
...           ...                       ...   
23429     12883.0             project_12317   
23430     21181.0             project_12057   
23431     13993.0              project_4461   
23432      5889.0               project_144   
23433      5529.0             project_16300   

                                            name_en  \
0                Winning Residence Sukhumvit Praksa   
1                                  Gift Condominium   
2                  Sanrak 9 Sukprayoon Nong Krathum   
3                                   Baan Sansabai 1   
4                        Baanpruksa Ratchapruek-345   
...                                             ...   
23429  Baan Klangmeung Urbanion Ladphrao-Chokchai 4   
23430                           @Home Kaset-Nawamin   
23431                    I-Design Wongwaen-Ramintra   
23432                 Koolpunt Ville 15 Park Avenue   
23433                          Thawisuk Garden Home   

                                             name_th propertytype_id  \
0                 วินนิ่ง เรสซิเดนท์ สุขุมวิท แพรกษา               1   
1                                กิ๊ฟท์ คอนโดมิเนียม               2   
2                     แสนรัก 9 ศุขประยูร หนองกระทุ่ม               3   
3                                      บ้านแสนสบาย 1               1   
4                             บ้านพฤกษา ราชพฤกษ์-345               3   
...                                              ...             ...   
23429  บ้านกลางเมือง เออร์บาเนี่ยน ลาดพร้าว-โชคชัย 4               3   
23430                          แอทโฮม เกษตร-นวมินทร์               3   
23431                     ไอ-ดีไซน์ วงแหวน-รามอินทรา               4   
23432                  กุลพันธ์วิลล์ 15 พาร์ค อเวนิว               1   
23433                             ทวีสุข การ์เด้นโฮม               1   

      propertytype_name_en propertytype_name_th  price_min     developer_id  \
0           Detached House                 บ้าน  3790000.0  developer_10678   
1                    Condo                คอนโด  1100000.0   developer_7319   
2                 Townhome             ทาวน์โฮม        0.0  developer_10510   
3           Detached House                 บ้าน        0.0  developer_10510   
4                 Townhome             ทาวน์โฮม  1930000.0   developer_1026   
...                    ...                  ...        ...              ...   
23429             Townhome             ทาวน์โฮม  5700000.0   developer_1760   
23430             Townhome             ทาวน์โฮม  3590000.0   developer_1793   
23431           Commercial         อาคารพาณิชย์  1870000.0   developer_2217   
23432       Detached House                 บ้าน  4000000.0   developer_1004   
23433       Detached House                 บ้าน  2800000.0  developer_10510   

                   developer_name_en  ... facility_meeting  facility_park  \
0       Winning Asset Plus Co., Ltd.  ...              NaN            NaN   
1           Gift Asset Land Co.,Ltd.  ...              NaN            1.0   
2                                NaN  ...              NaN            NaN   
3                                NaN  ...              NaN            NaN   
4       Preuksa Real Estate Co.,Ltd.  ...              NaN            1.0   
...                              ...  ...              ...            ...   
23429         AP (Thailand) Co,. Ltd  ...              NaN            1.0   
23430       Areeya Property Co.,Ltd.  ...              NaN            NaN   
23431  Wangthong Group Pub Co., Ltd.  ...              NaN            1.0   
23432     Homeland Service Co., Ltd.  ...              NaN            1.0   
23433                            NaN  ...              NaN            NaN   

       facility_playground facility_pool f

In [28]:
pro = project[['row_number','propertytype_name_th']]
pro

row_number propertytype_name_th
0          7880.0                 บ้าน
1         16689.0                คอนโด
2           290.0             ทาวน์โฮม
3         20719.0                 บ้าน
4           199.0             ทาวน์โฮม
...           ...                  ...
23429     12883.0             ทาวน์โฮม
23430     21181.0             ทาวน์โฮม
23431     13993.0         อาคารพาณิชย์
23432      5889.0                 บ้าน
23433      5529.0                 บ้าน

[23434 rows x 2 columns]

In [29]:
D_all= Rent_0.merge(pro, left_on='row_number', right_on='row_number')
D_all


row_number listing_district_name_th  volume_of_rental  \
0           1324                  จตุจักร               702   
1            136               บางกอกน้อย                33   
2            310                 ปากเกร็ด                22   
3           1570                   บางรัก               104   
4           1063                   พระนคร                11   
...          ...                      ...               ...   
1873        1585                บางคอแหลม                23   
1874         351                  ราชเทวี               103   
1875         694             เมืองนนทบุรี               100   
1876         633                  หลักสี่                14   
1877         899                  ลำลูกกา                23   

     propertytype_name_th  
0                    บ้าน  
1                ทาวน์โฮม  
2                ทาวน์โฮม  
3                    บ้าน  
4                   คอนโด  
...                   ...  
1873             ทาวน์โฮม  
1874              บ้านแฝด  
1875                คอนโด  
1876                 บ้าน  
1877                คอนโด  

[1878 rows x 4 columns]

In [30]:
ans_D_all = D_all.groupby('listing_district_name_th')[["volume_of_rental"]].max()
ans_D_all.sort_values('volume_of_rental',ascending=0)
#4. พื้นที่ ที่มีความต้องการเช่ามากที่สุด คือพื้นที่ใด และพื้นที่นั้นมีชนิดของบ้าน แบบไหนมากที่สุด (7 คะแนน)


volume_of_rental
listing_district_name_th                  
วัฒนา                                 3996
คลองเตย                               3340
ห้วยขวาง                              3270
เมืองสมุทรปราการ                      2177
บางรัก                                2167
เมืองนนทบุรี                          1541
ปทุมวัน                               1426
ราชเทวี                               1262
จตุจักร                               1250
สาทร                                  1061
พระนคร                                 927
พระโขนง                                863
คลองสาน                                837
บางนา                                  750
ปากเกร็ด                               604
ดินแดง                                 580
บางกะปิ                                554
ธนบุรี                                 548
บางพลี                                 536
สวนหลวง                                504
พญาไท                                  381
บางซื่อ                                321
ยานนาวา                                294
คลองหลวง                               265
บางเขน                                 248
จอมทอง                                 235
บางคอแหลม                              172
ประเวศ                                 169
บางพลัด                                162
บางบัวทอง                              142
ลาดกระบัง                              140
ภาษีเจริญ                              140
วังทองหลาง                             137
บึงกุ่ม                                121
บางแค                                  118
หลักสี่                                 98
บางใหญ่                                 96
คันนายาว                                96
บางบ่อ                                  90
พระประแดง                               87
บางกอกใหญ่                              75
บางกอกน้อย                              73
ราษฎร์บูรณะ                             69
พุทธมณฑล                                69
บางเสาธง                                67
ลาดพร้าว                                63
ธัญบุรี                                 58
ดอนเมือง                                49
มีนบุรี                                 38
บางขุนเทียน                             36
ลำลูกกา                                 31
สายไหม                                  26
เมืองนครปฐม                             20
เมืองปทุมธานี                           19
ทวีวัฒนา                                18
บางกรวย                                 18
สามพราน                                 17
ทุ่งครุ                                 12

In [31]:
B = D_all[['listing_district_name_th','propertytype_name_th']]
B

listing_district_name_th propertytype_name_th
0                     จตุจักร                 บ้าน
1                  บางกอกน้อย             ทาวน์โฮม
2                    ปากเกร็ด             ทาวน์โฮม
3                      บางรัก                 บ้าน
4                      พระนคร                คอนโด
...                       ...                  ...
1873                บางคอแหลม             ทาวน์โฮม
1874                  ราชเทวี              บ้านแฝด
1875             เมืองนนทบุรี                คอนโด
1876                  หลักสี่                 บ้าน
1877                  ลำลูกกา                คอนโด

[1878 rows x 2 columns]

In [32]:
V = B[B['listing_district_name_th']=='วัฒนา']
V


listing_district_name_th propertytype_name_th
15                      วัฒนา                 บ้าน
68                      วัฒนา             ทาวน์โฮม
75                      วัฒนา                คอนโด
94                      วัฒนา                คอนโด
188                     วัฒนา                คอนโด
269                     วัฒนา             ทาวน์โฮม
376                     วัฒนา             ทาวน์โฮม
383                     วัฒนา                 บ้าน
414                     วัฒนา                 บ้าน
501                     วัฒนา                 บ้าน
513                     วัฒนา                คอนโด
593                     วัฒนา              บ้านแฝด
616                     วัฒนา                คอนโด
625                     วัฒนา                 บ้าน
634                     วัฒนา             ทาวน์โฮม
684                     วัฒนา             ทาวน์โฮม
700                     วัฒนา             ทาวน์โฮม
741                     วัฒนา                 บ้าน
750                     วัฒนา                 บ้าน
766                     วัฒนา                คอนโด
769                     วัฒนา                คอนโด
787                     วัฒนา                คอนโด
802                     วัฒนา             ทาวน์โฮม
818                     วัฒนา             ทาวน์โฮม
821                     วัฒนา             ทาวน์โฮม
838                     วัฒนา                คอนโด
858                     วัฒนา                 บ้าน
980                     วัฒนา            โฮมออฟฟิศ
1035                    วัฒนา                 บ้าน
1050                    วัฒนา                 บ้าน
1114                    วัฒนา                คอนโด
1134                    วัฒนา                 บ้าน
1181                    วัฒนา             ทาวน์โฮม
1209                    วัฒนา                คอนโด
1236                    วัฒนา             ทาวน์โฮม
1262                    วัฒนา                คอนโด
1273                    วัฒนา                 บ้าน
1292                    วัฒนา         อาคารพาณิชย์
1337                    วัฒนา                คอนโด
1425                    วัฒนา                คอนโด
1486                    วัฒนา                 บ้าน
1578                    วัฒนา                 บ้าน
1595                    วัฒนา         อาคารพาณิชย์
1626                    วัฒนา                 บ้าน
1645                    วัฒนา                 บ้าน
1797                    วัฒนา             ทาวน์โฮม
1837                    วัฒนา                 บ้าน

In [33]:
a = V.groupby('propertytype_name_th').count()
a


listing_district_name_th
propertytype_name_th                          
คอนโด                                       14
ทาวน์โฮม                                    12
บ้าน                                        17
บ้านแฝด                                      1
อาคารพาณิชย์                                 2
โฮมออฟฟิศ                                    1

In [34]:
a.iloc[2:3]

listing_district_name_th
propertytype_name_th                          
บ้าน                                        17

**พื้นที่ที่มีความต้องการเช่ามากที่สุดคือ  วัฒนา จำนวนการเช่าอยู่ที่  3996 ที่ เเละมีเเบบบ้านมากที่สุดคือ บ้าน**

### 5. (เฉพาะ กรุงเทพมหานคร) เปรียบเทียบราคาโครงการที่แพงที่สุดในย่าน (subdistrict)ที่ราคาของอาหารการกินในช่วงกลางคืนที่ราคาสูงที่สุด (eating_nighttime_price) กับ ย่าน (subdistrict)ที่ราคาของอาหารการกินในช่วงกลางคืนที่ราคาต่ำที่สุด (eating_nighttime_price) (10 คะแนน)

In [35]:
project.head()

row_number                project_id                             name_en  \
0      7880.0  5ee08d64dda791001299b2a6  Winning Residence Sukhumvit Praksa   
1     16689.0             project_11326                    Gift Condominium   
2       290.0  62739c4089233e001362b001    Sanrak 9 Sukprayoon Nong Krathum   
3     20719.0              project_2010                     Baan Sansabai 1   
4       199.0             project_16838          Baanpruksa Ratchapruek-345   

                              name_th propertytype_id propertytype_name_en  \
0  วินนิ่ง เรสซิเดนท์ สุขุมวิท แพรกษา               1       Detached House   
1                 กิ๊ฟท์ คอนโดมิเนียม               2                Condo   
2      แสนรัก 9 ศุขประยูร หนองกระทุ่ม               3             Townhome   
3                       บ้านแสนสบาย 1               1       Detached House   
4              บ้านพฤกษา ราชพฤกษ์-345               3             Townhome   

  propertytype_name_th  price_min     developer_id  \
0                 บ้าน  3790000.0  developer_10678   
1                คอนโด  1100000.0   developer_7319   
2             ทาวน์โฮม        0.0  developer_10510   
3                 บ้าน        0.0  developer_10510   
4             ทาวน์โฮม  1930000.0   developer_1026   

              developer_name_en  ... facility_meeting  facility_park  \
0  Winning Asset Plus Co., Ltd.  ...              NaN            NaN   
1      Gift Asset Land Co.,Ltd.  ...              NaN            1.0   
2                           NaN  ...              NaN            NaN   
3                           NaN  ...              NaN            NaN   
4  Preuksa Real Estate Co.,Ltd.  ...              NaN            1.0   

   facility_playground facility_pool facility_security         date_created  \
0                  NaN           1.0               1.0  2020-06-10 14:36:04   
1                  NaN           1.0               1.0  2017-10-30 15:38:48   
2                  NaN           NaN               NaN  2022-05-05 16:43:28   
3                  NaN           NaN               NaN  2016-09-05 15:32:15   
4                  NaN           1.0               1.0  2019-09-05 11:49:37   

           date_finish         date_updated  source  \
0                  NaN  2020-12-04 10:32:58  baania   
1  2013-01-01 07:00:00  2020-12-04 10:49:00  baania   
2                  NaN  2022-05-05 17:30:03  baania   
3  1970-01-02 07:00:00  2020-12-04 10:45:31  baania   
4  1970-01-02 07:00:00  2021-03-17 12:00:04  baania   

                                         url_project  
0  https://baania.com/th/project/วินนิ่ง-เรสซิเดน...  
1  https://baania.com/th/project/กิ๊ฟท์-คอนโดมิเน...  
2  https://baania.com/th/project/แสนรัก-9-ศุขประย...  
3  https://baania.com/th/project/บ้านแสนสบาย-1-pr...  
4  https://baania.com/th/project/บ้านพฤกษา-ราชพฤก...  

[5 rows x 45 columns]

In [36]:
eat.head()

row_number  subdistrict_id subdistrict_name_en subdistrict_name_th  \
0           2            4001         HUAI KHWANG            ห้วยขวาง   
1         397            2632      THEP KRASATTRI         เทพกระษัตรี   
2          76            3509             PAK NAM              ปากน้ำ   
3         368            4628               SAMET               เสม็ด   
4         531            1166             TON PAO              ต้นเปา   

   district_id     district_name_en  district_name_th  province_id  \
0         4000          HUAI KHWANG          ห้วยขวาง         3781   
1         2630              THALANG              ถลาง         2616   
2         3499  MUEANG SAMUT PRAKAN  เมืองสมุทรปราการ         3498   
3         4617     MUEANG CHON BURI       เมืองชลบุรี         4616   
4         1164        SAN KAMPHAENG          สันกำแพง         1005   

  province_name_en province_name_th  eating_daytime  eating_daytime_price  \
0          Bangkok    กรุงเทพมหานคร            66.0                  72.0   
1           Phuket           ภูเก็ต            35.0                  37.0   
2     Samut Prakan      สมุทรปราการ            70.0                  84.0   
3        Chon Buri           ชลบุรี            68.0                  67.0   
4       Chiang Mai        เชียงใหม่            44.0                  36.0   

   eating_daytime_quality  eating_daytime_variety  eating_nighttime  \
0                    49.0                    92.0              75.0   
1                    30.0                    45.0              38.0   
2                    55.0                    79.0              65.0   
3                    56.0                    97.0              69.0   
4                    47.0                    51.0              47.0   

   eating_nighttime_price  eating_nighttime_quality  eating_nighttime_variety  \
0                    68.0                      54.0                      94.0   
1                    28.0                      28.0                      51.0   
2                    72.0                      57.0                      67.0   
3                    41.0                      54.0                      94.0   
4                    22.0                      61.0                      52.0   

   source         day  
0  baania  2022-08-31  
1  baania  2022-08-31  
2  baania  2022-08-31  
3  baania  2022-08-31  
4  baania  2022-08-31

In [37]:
project_2 = project[['row_number','name_th','price_min']]
eat_2 = eat[['row_number','subdistrict_name_th','province_name_th','eating_nighttime_price']]

In [38]:
all5= project_2.merge(eat_2, left_on='row_number', right_on='row_number')
all5


row_number                                    name_th  price_min  \
0         290.0             แสนรัก 9 ศุขประยูร หนองกระทุ่ม        0.0   
1         199.0                     บ้านพฤกษา ราชพฤกษ์-345  1930000.0   
2         270.0          โกลเด้น วัลเลย์ อัสสัมชัญ-ศรีราชา  2590000.0   
3          73.0                              โอฮาน่า คอนโด  1400000.0   
4         258.0                      ชโนทัย สุคนธสวัสดิ์ 8  6990000.0   
..          ...                                        ...        ...   
549       361.0                                  บ้านรัชกร  1700000.0   
550       192.0                        เดอะ พลูริส เขาใหญ่  3400000.0   
551       234.0                     บ้านเทพประทานพรนิเวศน์        0.0   
552       134.0               รุ่งอรุณ วิลล์ วัดส้มเกลี้ยง  4200000.0   
553       262.0  ลุมพินี เพลส แจ้งวัฒนะ - ปากเกร็ด สเตชั่น  2000000.0   

    subdistrict_name_th province_name_th  eating_nighttime_price  
0              แม่เหียะ        เชียงใหม่                    49.0  
1                บางจาก      สมุทรปราการ                    62.0  
2                 ดุสิต    กรุงเทพมหานคร                    72.0  
3              สำโรงใต้      สมุทรปราการ                    69.0  
4                 ลำพยา           นครปฐม                    60.0  
..                  ...              ...                     ...  
549              ท่าจีน        สมุทรสาคร                    48.0  
550           ลานตากฟ้า           นครปฐม                    61.0  
551           นิคมพัฒนา            ระยอง                    31.0  
552          คลองต้นไทร    กรุงเทพมหานคร                    68.0  
553          กระทุ่มราย    กรุงเทพมหานคร                    86.0  

[554 rows x 6 columns]

In [39]:
all5_BKK =all5[all5['province_name_th']=='กรุงเทพมหานคร']
all5_BKK

row_number                                      name_th  price_min  \
2         270.0            โกลเด้น วัลเลย์ อัสสัมชัญ-ศรีราชา  2590000.0   
10        422.0                           นิรันดร์ แกรนด์วิว   850000.0   
17        124.0  บ้านร็อคการ์เด้น แอร์พอร์ต สุวรรณภูมิ เฟส 1  2100000.0   
19        407.0                     บีอาร์ @ แอโร สุวรรณภูมิ  2500000.0   
23        111.0  บ้านเอื้ออาทร สุวรรณภูมิ 1 (วัดศรีวารีน้อย)   150000.0   
..          ...                                          ...        ...   
540       523.0                     ดิ โอโซน สวนเสือ-ศรีราชา  2450000.0   
544       214.0                       บ้านกลางเมือง พระราม 2  8019000.0   
545       540.0                             เดอะโฮม อ้อมค่าย  1890000.0   
552       134.0                 รุ่งอรุณ วิลล์ วัดส้มเกลี้ยง  4200000.0   
553       262.0    ลุมพินี เพลส แจ้งวัฒนะ - ปากเกร็ด สเตชั่น  2000000.0   

    subdistrict_name_th province_name_th  eating_nighttime_price  
2                 ดุสิต    กรุงเทพมหานคร                    72.0  
10                บางแค    กรุงเทพมหานคร                    81.0  
17              หนองจอก    กรุงเทพมหานคร                    84.0  
19             บางปะกอก    กรุงเทพมหานคร                    74.0  
23               ทับยาว    กรุงเทพมหานคร                    79.0  
..                  ...              ...                     ...  
540          บางนาเหนือ    กรุงเทพมหานคร                    77.0  
544        บางเชือกหนัง    กรุงเทพมหานคร                    76.0  
545         หนองค้างพลู    กรุงเทพมหานคร                    85.0  
552          คลองต้นไทร    กรุงเทพมหานคร                    68.0  
553          กระทุ่มราย    กรุงเทพมหานคร                    86.0  

[161 rows x 6 columns]

In [40]:
BKK_1 = all5_BKK.groupby('subdistrict_name_th')[['eating_nighttime_price','name_th','price_min','subdistrict_name_th']].max()
BKK_1.sort_values('eating_nighttime_price',ascending=0)
#ราคาที่อยู่ในย่านลำต้อยติ่ง เเพงที่สุด เเละอยู่ในโครงการ วิลล่า สราสินี รีสอร์ท แอท โฮม เป็นโครงการที่เเพงที่สุด เเละ ราคาที่อยู่ในย่านลุมพินี ต่ำที่สุด เเละอยู่ในโครงการ บ้านกลางเมือ สวิสทาวน์

eating_nighttime_price  \
subdistrict_name_th                           
ลำต้อยติ่ง                            100.0   
ทุ่งครุ                                87.0   
กระทุ่มราย                             86.0   
หนองแขม                                85.0   
หนองค้างพลู                            85.0   
...                                     ...   
ทุ่งมหาเมฆ                             54.0   
คลองเตย                                54.0   
บางรัก                                 52.0   
คลองตัน                                51.0   
ลุมพินี                                46.0   

                                                       name_th  price_min  \
subdistrict_name_th                                                         
ลำต้อยติ่ง                      วิลล่า สราสินี รีสอร์ท แอท โฮม  5890000.0   
ทุ่งครุ                                              ทิพาคาร 1   427000.0   
กระทุ่มราย           ลุมพินี เพลส แจ้งวัฒนะ - ปากเกร็ด สเตชั่น  2000000.0   
หนองแขม                                   บ้านกลางสวน บางละมุง  1700000.0   
หนองค้างพลู                                   เดอะโฮม อ้อมค่าย  1890000.0   
...                                                        ...        ...   
ทุ่งมหาเมฆ                                  เฟื่องฟ้า คอนโดเทล   890000.0   
คลองเตย                                   เดอะ ซีซั่น วัลเล่ย์  1950000.0   
บางรัก                         เอ็กซ์เท็น คอนโดมิเนียม ราชมงคล  1200000.0   
คลองตัน                      บ้านพฤกษา 62 (เทพารักษ์-กิ่งแก้ว)   899000.0   
ลุมพินี                                บ้านกลางเมือง สวิสทาวน์  4500000.0   

                    subdistrict_name_th  
subdistrict_name_th                      
ลำต้อยติ่ง                   ลำต้อยติ่ง  
ทุ่งครุ                         ทุ่งครุ  
กระทุ่มราย                   กระทุ่มราย  
หนองแขม                         หนองแขม  
หนองค้างพลู                 หนองค้างพลู  
...                                 ...  
ทุ่งมหาเมฆ                   ทุ่งมหาเมฆ  
คลองเตย                         คลองเตย  
บางรัก                           บางรัก  
คลองตัน                         คลองตัน  
ลุมพินี                         ลุมพินี  

[157 rows x 4 columns]

### 6. แสดงย่านที่ อาหารการกินมีความหลากหลายที่สุดในช่วงกลางวัน (eating_daytime_variety) และ กลางคืน (eating_nighttime_variety) ของแต่ละจังหวัด (13 คะแนน)

In [41]:
eat

row_number  subdistrict_id subdistrict_name_en subdistrict_name_th  \
0             2            4001         HUAI KHWANG            ห้วยขวาง   
1           397            2632      THEP KRASATTRI         เทพกระษัตรี   
2            76            3509             PAK NAM              ปากน้ำ   
3           368            4628               SAMET               เสม็ด   
4           531            1166             TON PAO              ต้นเปา   
..          ...             ...                 ...                 ...   
549         116            3726          THA TAMNAK           ท่าตำหนัก   
550         151            3476            CHAK BOK               ชากบก   
551         290            1015             MAE HIA            แม่เหียะ   
552          69            1020      NONG PA KHRANG        หนองป่าครั่ง   
553          35            1165          CHAE CHANG             แช่ช้าง   

     district_id     district_name_en  district_name_th  province_id  \
0           4000          HUAI KHWANG          ห้วยขวาง         3781   
1           2630              THALANG              ถลาง         2616   
2           3499  MUEANG SAMUT PRAKAN  เมืองสมุทรปราการ         3498   
3           4617     MUEANG CHON BURI       เมืองชลบุรี         4616   
4           1164        SAN KAMPHAENG          สันกำแพง         1005   
..           ...                  ...               ...          ...   
549         3719       NAKHON CHAI SI         นครชัยศรี         3667   
550         3474             BAN KHAI          บ้านค่าย         3431   
551         1006    MUEANG CHIANG MAI    เมืองเชียงใหม่         1005   
552         1006    MUEANG CHIANG MAI    เมืองเชียงใหม่         1005   
553         1164        SAN KAMPHAENG          สันกำแพง         1005   

    province_name_en province_name_th  eating_daytime  eating_daytime_price  \
0            Bangkok    กรุงเทพมหานคร            66.0                  72.0   
1             Phuket           ภูเก็ต            35.0                  37.0   
2       Samut Prakan      สมุทรปราการ            70.0                  84.0   
3          Chon Buri           ชลบุรี            68.0                  67.0   
4         Chiang Mai        เชียงใหม่            44.0                  36.0   
..               ...              ...             ...                   ...   
549    Nakhon Pathom           นครปฐม            65.0                  68.0   
550           Rayong            ระยอง            44.0                  42.0   
551       Chiang Mai        เชียงใหม่            66.0                  58.0   
552       Chiang Mai        เชียงใหม่            69.0                  58.0   
553       Chiang Mai        เชียงใหม่            34.0                  38.0   

     eating_daytime_quality  eating_daytime_variety  eating_nighttime  \
0                      49.0                    92.0              75.0   
1                      30.0                    45.0              38.0   
2                      55.0                    79.0              65.0   
3                      56.0                    97.0              69.0   
4                      47.0                    51.0              47.0   
..                      ...                     ...               ...   
549                    56.0                    76.0              64.0   
550                    53.0                    27.0              38.0   
551                    68.0                    72.0              64.0   
552                    67.0                    91.0              72.0   
553                    35.0                    26.0              32.0   

     eating_nighttime_price  eating_nighttime_quality  \
0                      68.0                      54.0   
1                      28.0                      28.0   
2                      72.0                      57.0   
3                      41.0                      54.0   
4                      22.0                      61.0   
..                      ...                       ...   
549           

In [42]:
Eat_province = eat[['province_name_th','subdistrict_name_th','eating_daytime_variety','eating_nighttime_variety']]
Eat_province

province_name_th subdistrict_name_th  eating_daytime_variety  \
0      กรุงเทพมหานคร            ห้วยขวาง                    92.0   
1             ภูเก็ต         เทพกระษัตรี                    45.0   
2        สมุทรปราการ              ปากน้ำ                    79.0   
3             ชลบุรี               เสม็ด                    97.0   
4          เชียงใหม่              ต้นเปา                    51.0   
..               ...                 ...                     ...   
549           นครปฐม           ท่าตำหนัก                    76.0   
550            ระยอง               ชากบก                    27.0   
551        เชียงใหม่            แม่เหียะ                    72.0   
552        เชียงใหม่        หนองป่าครั่ง                    91.0   
553        เชียงใหม่             แช่ช้าง                    26.0   

     eating_nighttime_variety  
0                        94.0  
1                        51.0  
2                        67.0  
3                        94.0  
4                        52.0  
..                        ...  
549                      61.0  
550                      17.0  
551                      70.0  
552                      89.0  
553                      19.0  

[554 rows x 4 columns]

In [50]:
Eat_day = Eat_province.groupby('province_name_th')[['eating_daytime_variety','subdistrict_name_th']].max()
Eat_day.sort_values('eating_daytime_variety',ascending=0) #กลางวัน

eating_daytime_variety subdistrict_name_th
province_name_th                                            
กรุงเทพมหานคร                      100.0              โคกแฝด
ระยอง                              100.0           แม่น้ำคู้
เชียงใหม่                          100.0             ไชยสถาน
ชลบุรี                              99.0          ไร่หลักทอง
ภูเก็ต                              99.0              ไม้ขาว
นครปฐม                              93.0              ไร่ขิง
นนทบุรี                             91.0              ไทรม้า
ปทุมธานี                            90.0        เชียงรากใหญ่
สมุทรปราการ                         84.0      ในคลองบางปลากด
สมุทรสาคร                           82.0              โคกขาม

In [52]:
Eat_night= Eat_province.groupby('province_name_th')[['eating_nighttime_variety','subdistrict_name_th']].max()
Eat_night.sort_values('eating_nighttime_variety',ascending=0) #กลางคืน

eating_nighttime_variety subdistrict_name_th
province_name_th                                              
กรุงเทพมหานคร                        100.0              โคกแฝด
ระยอง                                100.0           แม่น้ำคู้
ชลบุรี                                99.0          ไร่หลักทอง
เชียงใหม่                             98.0             ไชยสถาน
ภูเก็ต                                97.0              ไม้ขาว
นนทบุรี                               96.0              ไทรม้า
นครปฐม                                94.0              ไร่ขิง
ปทุมธานี                              89.0        เชียงรากใหญ่
สมุทรสาคร                             85.0              โคกขาม
สมุทรปราการ                           82.0      ในคลองบางปลากด